sympy/solvers/diophantine.py

```diff
@@ -152,11 +152,24 @@ def diophantine(eq, param=symbols("t", integer=True), syms=None):
             if not is_sequence(syms):
                 raise TypeError(
                     'syms should be given as a sequence, e.g. a list')
-            syms = [i for i in syms if i in var]
+
             if syms != var:
```

Comment:
Since you've made `var` a list, you should make `syms` a list, too, or else if a tuple is sent it will trivially fail this test.